### ISSUE TITLE
Problems with Python 2.7 encoding and commandlines using codepage 850

### ISSUE THREAD

**Issue by BenchMasterFlash** [conan-io/conan#984]
Hi, in my `conanfile.py` I wrapped a compiler call with your `self.run(...)`-method. That works great until my compiler printed some warnings, that contained some German umlauts like ä or ö.

Here's the error-code I got:
```ERROR: Unable to build it successfully
  File "D:\my\project\conanfile.py", line 70, in build
    self.run(compiler, args)
  File "C:\Python27\lib\site-packages\conans\model\conan_file.py", line 235, in run
    retcode = self._runner(command, output, os.path.abspath(RUN_LOG_NAME),  cwd)
  File "C:\Python27\lib\site-packages\conans\client\runner.py", line 44, in __call__
    return self._pipe_os_call(command, stream_output, None, cwd)
  File "C:\Python27\lib\site-packages\conans\client\runner.py", line 66, in _pipe_os_call
    get_stream_lines(proc.stdout)
  File "C:\Python27\lib\site-packages\conans\client\runner.py", line 60, in get_stream_lines
    stream_output.write(decoded_line)
  File "C:\Python27\lib\site-packages\colorama\ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "C:\Python27\lib\site-packages\colorama\ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "C:\Python27\lib\site-packages\colorama\ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "C:\Python27\lib\site-packages\colorama\ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
  File "C:\Python27\lib\encodings\cp850.py", line 12, in encode
    return codecs.charmap_encode(input,errors,encoding_map)
UnicodeEncodeError: 'charmap' codec can't encode character u'\u201e' in position 6: character maps to <undefined>
```

Unfortunately, I cant get the compiler to print english warnings and errors and work around this problem. I'm working with German Windows 7 and my console is using codepage 850. I did **not** set the environment variable `PYTHONIOENCODING`. If I set it locally to `Windows-1252` everything works fine. But I can't ensure that it's set on any other system.

**Comment by memsharded** [comment]
You mean py2.7, right? From the trace it seems so.

**Comment by BenchMasterFlash** [comment]
Yeah, it's Python 2.7.13, sorry for confusion. Seems like a typo in the topic.

**Comment by memsharded** [comment]
Hi!
I have been working on this for a while. As usual with encodings, a total nightmare. I came with a patch in: https://github.com/conan-io/conan/pull/1023 that does aggressive encoding/decoding to latin-1, will loose chars, but at least might work without crashing. As you say, codepage 850, is very common, so this is expected to fail otherwise for more people that do not set the encoding.

Actually, the umlauts where not the one failing but the u'\u201e', which is some double low quotes. 

Hope this fixes the issue. Thanks again for reporting

**Comment by memsharded** [comment]
Released in 0.20, please upgrade and report if something fails again. Thanks!
